### PR TITLE
Migrate to null-safety

### PR DESCRIPTION
**Pull request by xvrh**
Hi, this is my attempt to convert this package to null-safety.

### CODE REVIEW

**Comment by xvrh** [comment]
I also replaced Travis with Github Actions (since Travis was not triggering anymore?).
It doesn't trigger for this pull request (for security reason I suppose), but it executed correctly on my [fork](https://github.com/xvrh/analysis_server_lib/pull/1)

**Review by devoncarew — APPROVED** [review]
Wow, this is great - very thorough! I'm curious how many iterations it took for the code generator to generate good null safe code.

One minor nit in the github file. I'll likely land this and do some local testing.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - uses: dart-lang/setup-dart@v0.3
+        with:
+          sdk: ${{ matrix.sdk }}
+      - uses: actions/checkout@v2
+      - run: dart --version
+      - run: dart pub get
+      # Verify the library re-generates.
+      - run: dart --enable-asserts tool/generate_analysis.dart
+      # Verify that the libraries are error free.
+      - run: dart analyze --fatal-infos --fatal-warnings .
```
[the remark this answers is not in the document]
Comment (xvrh) [reply .github/workflows/build.yaml]: Done

**Comment by xvrh** [comment]
Thanks for the approval.

I think we will probably find some issues (like an unexpected null value in a json response). But since all the tests passed, I wanted to already submit this. It will be good occasion to increase the test coverage.

**Pull request merged**